File: loc_phat_2.cpp
#include<iostream>
#include<string>

using namespace std;

int main(){
    int t;
    cin>>t;
    while(t--){
        int n;
        cin>>n;
        int sum=0, mul=2;
        for(int i=1;i<=n;i++, mul*=2){
            sum+=mul;
        }
        cout<<sum<<endl;
        for(int i=1;i<=n;i++){
            string res="", last="";
            for(int j=0;j<i;j++){
                res+='6';
                last+='8';
            }
            cout<<res<<' ';
            while(res!=last){
                for(int j=i-1;j>=0;j--){
                    if(res[j]=='6'){                              
                        res[j]='8';                                 
                        break;
                    }
                    if(res[j]='8'){
                        res[j]='6';
                        continue;
                    }
                }
                cout<<res<<' ';
            }
        }
        cout<<endl;
    }
    return 0;
}

File: loc_phat_2.py
t=int(input())

for v in range(t):
    n=int(input())
    print(2**n)
    res, last='',''
    for j in range(n):
        res+='6'
        last+='8'
    print(res,end=' ')
    while res!=last:
        for j in range(n-1,-1,-1):
            if(res[j]=='6'):
                tmp=list(res)
                tmp[j]='8'
                res=''.join(tmp)
                break
            if(res[j]=='8'):
                tmp=list(res)
                tmp[j]='6'
                res=''.join(tmp)
                continue
        print(res,end=' ')
    print()